packages/glimmer-test-helpers/lib/environment.ts

```diff
@@ -232,9 +235,13 @@ export class EmberishCurlyComponent extends GlimmerObject {
   public attrs: Attrs;
   public element: Element;
   public parentView: Component = null;
+  public args: ProcessedArgs;
+  public argsRevision: number;
 
   static create(args: { attrs: Attrs }): EmberishCurlyComponent {
-    return super.create(args) as EmberishCurlyComponent;
+    let result = super.create(args) as EmberishCurlyComponent;
+    result.argsRevision = result.args && result.args.tag && result.args.tag.value();
```

Comment:
Yeah, about half way in to writing this code I realized that it's basically all recomputed all the time.